### ISSUE TITLE
Obfuscation breaking with two lambdas

### ISSUE THREAD

**Issue by Nukiloco** [cherriesandmochi/gdmaim#13]
It seems like when obfuscating code, having code with something on the lines of this:
```gdscript
var lambda1 = func():
	# "test value" here will not get obfuscated
	var test_value = some_global_class.fruits.map(func(a): return a)
	# however "test value" here will get obfuscated, which will break the project
	print(test_value)
```
will break the code

**Comment by Nukiloco** [comment]
Even with turning off the obfuscation for the line, it seems like it gets ignored and any variable containing that variable name still gets obfuscated.

**Comment by cherriesandmochi** [comment]
Thanks! Nested lambdas don't work very well at the moment. I do already have a rework for the way scope is detected in mind, which should fix issues like this, but haven't started working on it yet.

**Comment by Nukiloco** [comment]
Here is a reproduction project for the issue!
[gdmaim_lambda_issue_1.zip](https://github.com/cherriesandmochi/gdmaim/files/14797944/gdmaim_lambda_issue_1.zip)

**Comment by cherriesandmochi** [comment]
Fixed by 7faaac7.
Thanks for the bug report!
